core/app/components/forms/elements/ContextEditorElement.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import type { Node } from "prosemirror-model";
+
+import { useMemo } from "react";
+import { Value } from "@sinclair/typebox/value";
+import { useFormContext } from "react-hook-form";
+import { richTextInputConfigSchema } from "schemas";
+
+import { FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+
+import type { ElementProps } from "../types";
+import { ContextEditorClient } from "../../ContextEditorClient";
+import { useFormElementToggleContext } from "../FormElementToggleContext";
+
+const EditorFormElement = ({
+	label,
+	help,
+	onChange,
+	initialValue,
+}: {
+	label: string;
+	help?: string;
+	onChange: (state: any) => void;
+	initialValue?: Node;
+}) => {
+	const memoEditor = useMemo(() => {
```

Comment:
I'm memoizing liberally—I've been falling into infinite render traps and this seems to help, but I couldn't exactly tell you why